mlflow/store/artifact/artifact_repo.py

```diff
@@ -384,14 +384,14 @@ def upload_trace_data(self, trace_data: str) -> None:
 def write_local_temp_trace_data_file(trace_data: str):
     with tempfile.TemporaryDirectory() as temp_dir:
         temp_file = Path(temp_dir, TRACE_DATA_FILE_NAME)
-        temp_file.write_text(trace_data)
+        temp_file.write_text(trace_data, encoding="utf-8")
```

Comment:
In Python 3.15, UTF8 will be the default: https://peps.python.org/pep-0686/

Reply:
We write or read everywhere without explicitly specifying `encoding`. This is a short-term solution to close #14243.

Reply:
nit: shall we define the common encoding (utf8) as a const variable?

Reply:
Confirmed that the default encoding here is system dependent, the change looks good. https://docs.python.org/3/library/functions.html#open